flexmeasures/data/models/reporting/tests/test_aggregator.py

```diff
@@ -157,7 +157,9 @@ def test_resampling(setup_dummy_data):
 
 
 def test_multiple_sources(setup_dummy_data):
-    """check that it uses the first source as default, instead of using all of them."""
+    """check that it uses the first source as default, instead of using all of them.
```

Comment:
But it is no longer true that we'd take the first source as a default. And also, this test doesn't actually check that. My expectation is that:
- we take the most recent belief per event per source, for example:
  - at 1 PM, source A believed X would happen at 5 PM
  - at 1 PM, source B believed Y would happen at 6 PM
  - at 2 PM, source A believed Z would happen at 6 PM
- we then take the first source per event (I'm not 100% sure whether that would lead to taking the oldest or newest belief, in case the sources do not share a belief time, but I believe it would take the oldest belief, given that the `belief_time` index is sorted ascending in [`timely_beliefs.BeliefsDataFrame.__init__`](https://github.com/SeitaBV/timely-beliefs/blob/e69478849cd14925c7498e6b6a078ef62acfe299/timely_beliefs/beliefs/classes.py#L884))
  - at 1 PM, source A believed X would happen at 5 PM
  - at 1 PM, source B believed Y would happen at 6 PM

So the resulting frame could still have multiple sources, but just 1 source per event.

Reply:
That makes sense. 

 I would avoid having mixed sources and go back to the previous approach of filtering by the first source, by default. For instance, this could be an issue when computing the aggregated power of a group of devices including a scheduled device. We could get actual power commands and scheduled power commands mixed up.



An alternative is to rise an exception when `len(sources) > 1` and the source is not specified in the input.

Reply:
I like this alternative. Is it already possible to set two sources to cover the case where one source takes over from another, and the reporter may need data from both during the period of transitioning?

Reply:
Good point!

I've just pushed some changes to support this. Let me know what you think :smile:

Reply:
I like the new test, thank you! What's left is only the check that would raise, right?

Reply:
What about the `raise` statement?

Reply:
But what about the case you brought up?

> For instance, this could be an issue when computing the aggregated power of a group of devices including a scheduled device. We could get actual power commands and scheduled power commands mixed up.

If no sources are explicitly set, it seems to me that this (mixing up schedules and measurements) could silently happen? I thought we wanted to raise in that situation, to force the user to specify sources explicitly.

Reply:
Yes, you're right! I've updated the reporter and added an extra test case to cover this.